upup/pkg/fi/k8sapi/k8s_keystore.go

```diff
@@ -90,22 +90,23 @@ func (c *KubernetesKeystore) findSecret(id string) (*v1.Secret, error) {
 	return secret, nil
 }
 
-func (c *KubernetesKeystore) FindKeypair(id string) (*pki.Certificate, *pki.PrivateKey, error) {
+func (c *KubernetesKeystore) FindKeypair(id string) (*pki.Certificate, *pki.PrivateKey, string, error) {
 	secret, err := c.findSecret(id)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, "", err
 	}
 
 	if secret == nil {
-		return nil, nil, nil
+		return nil, nil, "", nil
 	}
 
 	keypair, err := ParseKeypairSecret(secret)
 	if err != nil {
-		return nil, nil, fmt.Errorf("error parsing secret %s/%s from kubernetes: %v", c.namespace, id, err)
+		return nil, nil, "", fmt.Errorf("error parsing secret %s/%s from kubernetes: %v", c.namespace, id, err)
 	}
 
-	return keypair.Certificate, keypair.PrivateKey, nil
+	// TODO: KubernetesKeystore does not seem to be using the newer version of keypairs.
```

Comment:
Not sure what you mean.  keyset.yaml doesn't make sense here - keyset.yaml makes a filesystem behave more like an API server.   So we don't need keyset.yaml for an API-server-backed key store.